app/javascript/mastodon/components/dropdown_selector.tsx

```diff
@@ -0,0 +1,185 @@
+
+  const handleDocumentClick = useCallback(
+    (e: MouseEvent | TouchEvent) => {
+      if (
+        nodeRef.current &&
+        e.target instanceof Node &&
+        !nodeRef.current.contains(e.target)
+      ) {
+        onClose();
+        e.stopPropagation();
+      }
+    },
+    [nodeRef, onClose],
+  );
+
+  const handleClick = useCallback(
+    (
+      e: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>,
+    ) => {
+      const value = e.currentTarget.getAttribute('data-index');
+
+      e.preventDefault();
+
+      onClose();
+      if (value) onChange(value);
+    },
+    [onClose, onChange],
+  );
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLLIElement>) => {
+      const value = e.currentTarget.getAttribute('data-index');
+      const index = items.findIndex((item) => item.value === value);
+
+      let element: Element | null | undefined = null;
+
+      switch (e.key) {
+        case 'Escape':
+          onClose();
+          break;
+        case ' ':
+        case 'Enter':
+          handleClick(e);
+          break;
+        case 'ArrowDown':
+          element =
+            nodeRef.current?.children[index + 1] ??
+            nodeRef.current?.firstElementChild;
+          break;
+        case 'ArrowUp':
+          element =
+            nodeRef.current?.children[index - 1] ??
+            nodeRef.current?.lastElementChild;
+          break;
+        case 'Tab':
+          if (e.shiftKey) {
+            element =
+              nodeRef.current?.children[index + 1] ??
+              nodeRef.current?.firstElementChild;
+          } else {
+            element =
+              nodeRef.current?.children[index - 1] ??
+              nodeRef.current?.lastElementChild;
+          }
+          break;
+        case 'Home':
+          element = nodeRef.current?.firstElementChild;
+          break;
+        case 'End':
+          element = nodeRef.current?.lastElementChild;
+          break;
+      }
+
+      if (element && element instanceof HTMLElement) {
+        const selectedValue = element.getAttribute('data-index');
+        element.focus();
+        if (selectedValue) setCurrentValue(selectedValue);
+        e.preventDefault();
+        e.stopPropagation();
+      }
+    },
+    [nodeRef, items, onClose, handleClick, setCurrentValue],
+  );
+
+  useEffect(() => {
+    document.addEventListener('click', handleDocumentClick, { capture: true });
+    document.addEventListener('touchend', handleDocumentClick, listenerOptions);
+    focusedItemRef.current?.focus({ preventScroll: true });
+
+    return () => {
+      document.removeEventListener('click', handleDocumentClick, {
+        capture: true,
+      });
+      document.removeEventListener(
+        'touchend',
+        handleDocumentClick,
+        listenerOptions,
+      );
+    };
+  }, [handleDocumentClick]);
+
+  return (
+    <ul style={{ ...style }} role='listbox' ref={nodeRef}>
```

Comment:
Could be `style={style}`

Reply:
Maybe we could also add `className={className}` here? Otherwise rename it to `classNamePrefix` if it is only used for the options?